### ISSUE TITLE
Other *.NPK title support?

### ISSUE THREAD

**Issue by Kojichao** [morkt/GARbro#68]
Hello there!

I stumbled across this awesome program just a little while ago, and I'm so glad and thankful that it's a GUI. A LOT of Visual Novel Archive tools are all command line, so it's really nice to see some forays into the GUI direction.

Anywho, a lot of the Visual Novels that I'm looking at are different to the usual Eroge's, and I was wondering if it's possible to add support for another Nitro+ title, the new Windows 10 re-release of Togainu no Chi. It is a Nitro+CHiRAL title, but it's using the same encryption style as Sonicomi and TokyoNecro. (Just obviously a different title...).

I had a gander at the source for the GARbro, and didn't see any immediate way to add support for extra games ourselves very easily (And I am having a dilly of a time trying to locate the extra API/DLL files for references in the solution), so I figured I'd ask.

Thank you very much for your consideration!

(Also, if you need the script.npk file for testing, am I allowed to attach it here or should I send it somewhere?)

**Comment by morkt** [comment]
script.npk is of no use, archive key is embedded into executable file. currently i have no access to windows 10 release, and considering this VN has much easier accessible earlier releases, this would be on low priority for a while.

**Comment by Kojichao** [comment]
Ahh, that's good to know! o.o  I wasn't completely aware of that. And now that I am, I can also present to you the EXE. ^^ 

But I know you said you are making this a low priority. I'm just glad to know I got a response and that this could be a potential possibility. ^^

**Comment by rockleevk** [comment]
I know that already someone got an answer on this matter, but wanted to report as well that my recently purchased _Saya no Uta Nitro The Best! Vol.2 - Windows 10 Support Edition_ have same issue, cannot open at all the .NPK (NPK2 type as I saw when I tried to open with hex editor) files to extract the script file or system file. 

Can upload game files without any problem if needed here.

Thanks a lot for your efforts!

**Comment by lsssk11331** [comment]
just wanted to follow up and ask if there will be progress on this? will attach some files in case only:
[GDrive](https://drive.google.com/file/d/1pS0m3NXv6WdyHTyk1Tg4TPLjvuhVMhoR/view?usp=sharing)

**Comment by i4ri2n8t** [comment]
Game Request: Saya no Uta Nitro The Best! Vol.2 - Windows 10 Support Edition (Japanese)
EXE: [https://mega.nz/#!P90RVAKK!U-jh1XHe1mtpUwb6-jqUrfIc17SJgd3nTm59k6B8Bvc](https://mega.nz/#!P90RVAKK!U-jh1XHe1mtpUwb6-jqUrfIc17SJgd3nTm59k6B8Bvc)

Game Request: The Song of Saya (Steam)
EXE: [https://mega.nz/#!P90RVAKK!U-jh1XHe1mtpUwb6-jqUrfIc17SJgd3nTm59k6B8Bvc](https://mega.nz/#!bhkT2CSJ!kom4ZH63EuCMlb5P9leeLQunyHyUUf4-f1Ux5UoDBbg
)

Both are NPK, but cannot be deployed with the current GARBro. I would appreciate it if you could respond.

**Comment by Pierrounator** [comment]
I would love to see a Togainu no Chi and Sweet Pool support *_*

We are translating it in french, there is no official BL game in french. We already did most of the translation with the fan patch. But we would like to use the 2020 version which is **.npk** files.

**Comment by Pierrounator** [comment]
Btw, I have the keys for both games. I can extract many .npk but not the cg...

**Comment by nicefunfungal** [comment]
I need some help with this myself now for YOU and ME and HER..

**Comment by aikoofujimotoo** [comment]
> I need some help with this myself now for YOU and ME and HER..

It's NPK3. So far no unpackers can unpack npk3
![image](https://user-images.githubusercontent.com/35658068/85149312-1a9ec600-b28c-11ea-8f12-daa6808d1e84.png)

**Comment by cimo95** [comment]
> > I need some help with this myself now for YOU and ME and HER..
> 
> It's NPK3. So far no unpackers can unpack npk3
> ![image](https://user-images.githubusercontent.com/35658068/85149312-1a9ec600-b28c-11ea-8f12-daa6808d1e84.png)

even 5 months past, none can unpack NPK3 :(

**Comment by Pierrounator** [comment]
I did it, I achieved to unpack NPK3 files by myself, now I need to create the “repacker”

**Comment by RabbidHype** [comment]
Oh sweet!

**Comment by PitchRE** [comment]
> I did it, I achieved to unpack NPK3 files by myself, now I need to create the “repacker”
@Pierrounator 
any progress?

**Comment by Anna-Hardy** [comment]
Hello! I'm here bc of the remake of Dramatical Murder, and I want to unpack the npk files so I can port the Russian translation from the original game that had npa files to the remaster, So i'm in look for an npk extractor and if possible repacker as well. This is the closest thing I have found so far. @Pierrounator did you finally finish the tool? I would be grateful for any answer. Thanks in advance.

**Comment by Pierrounator** [comment]
> Hello! I'm here bc of the remake of Dramatical Murder, and I want to unpack the npk files so I can port the Russian translation from the original game that had npa files to the remaster, So i'm in look for an npk extractor and if possible repacker as well. This is the closest thing I have found so far. [@Pierrounator](https://github.com/Pierrounator) did you finally finish the tool? I would be grateful for any answer. Thanks in advance.

Yup we achieved to unpack and repack it after editing the images, we also tried with Slow Damage and it works fine but the “tool” isn't compiled. It's not a .exe and I'm not sure my friend who did it would like to share, I'm gonna ask him.

**Comment by Anna-Hardy** [comment]
> > Hello! I'm here bc of the remake of Dramatical Murder, and I want to unpack the npk files so I can port the Russian translation from the original game that had npa files to the remaster, So i'm in look for an npk extractor and if possible repacker as well. This is the closest thing I have found so far. [@Pierrounator](https://github.com/Pierrounator) did you finally finish the tool? I would be grateful for any answer. Thanks in advance.
> 
> Yup we achieved to unpack and repack it after editing the images, we also tried with Slow Damage and it works fine but the “tool” isn't compiled. It's not a .exe and I'm not sure my friend who did it would like to share, I'm gonna ask him.

Hello again! OMG that's great! If you could please get him to try to compile it and share it, it would be so great! I'm up to paying like 5 to 15-20 euro if that's needed tbh bc I'm kind of desperate 😭 Again, thanks so much in advance! Kisses!

>

**Comment by Anna-Hardy** [comment]
> > Hello! I'm here bc of the remake of Dramatical Murder, and I want to unpack the npk files so I can port the Russian translation from the original game that had npa files to the remaster, So i'm in look for an npk extractor and if possible repacker as well. This is the closest thing I have found so far. [@Pierrounator](https://github.com/Pierrounator) did you finally finish the tool? I would be grateful for any answer. Thanks in advance.
> 
> Yup we achieved to unpack and repack it after editing the images, we also tried with Slow Damage and it works fine but the “tool” isn't compiled. It's not a .exe and I'm not sure my friend who did it would like to share, I'm gonna ask him.


Sorry to write again, but a lot of time has passed and you still didn't answer. I really hope yall could provide the tool at least for personal use. Thanks in advance yet again.

**Comment by Pierrounator** [comment]
> > > Hello! I'm here bc of the remake of Dramatical Murder, and I want to unpack the npk files so I can port the Russian translation from the original game that had npa files to the remaster, So i'm in look for an npk extractor and if possible repacker as well. This is the closest thing I have found so far. [@Pierrounator](https://github.com/Pierrounator) did you finally finish the tool? I would be grateful for any answer. Thanks in advance.
> > 
> > 
> > Yup we achieved to unpack and repack it after editing the images, we also tried with Slow Damage and it works fine but the “tool” isn't compiled. It's not a .exe and I'm not sure my friend who did it would like to share, I'm gonna ask him.
> 
> Sorry to write again, but a lot of time has passed and you still didn't answer. I really hope yall could provide the tool at least for personal use. Thanks in advance yet again.

Yeah I forgot to ask him, but I did and this is what he sent me back 
<img width="788" height="164" alt="Image" src="https://github.com/user-attachments/assets/fb88fe6f-e2ef-4145-bcec-c2cc8021c786" />


lit. means: `“you share it, I kill you :)”` in french

**Comment by Anna-Hardy** [comment]
> > > > Hello! I'm here bc of the remake of Dramatical Murder, and I want to unpack the npk files so I can port the Russian translation from the original game that had npa files to the remaster, So i'm in look for an npk extractor and if possible repacker as well. This is the closest thing I have found so far. [@Pierrounator](https://github.com/Pierrounator) did you finally finish the tool? I would be grateful for any answer. Thanks in advance.
> > > 
> > > 
> > > Yup we achieved to unpack and repack it after editing the images, we also tried with Slow Damage and it works fine but the “tool” isn't compiled. It's not a .exe and I'm not sure my friend who did it would like to share, I'm gonna ask him.
> > 
> > 
> > Sorry to write again, but a lot of time has passed and you still didn't answer. I really hope yall could provide the tool at least for personal use. Thanks in advance yet again.
> 
> Yeah I forgot to ask him, but I did and this is what he sent me back <img alt="Image" width="788" height="164" src="https://private-user-images.githubusercontent.com/14140656/497375077-fb88fe6f-e2ef-4145-bcec-c2cc8021c786.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk5MTgxMjksIm5iZiI6MTc1OTkxNzgyOSwicGF0aCI6Ii8xNDE0MDY1Ni80OTczNzUwNzctZmI4OGZlNmYtZTJlZi00MTQ1LWJjZWMtYzJjYzgwMjFjNzg2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDA4VDEwMDM0OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThhMzE5NTUxMjJmZGE0NjFiYTgxZGNiYWI5MjE3Mjg1NGFlN2NiOWNhODFlNDNjYTMzNDYzMzU0NzJkYzI0MjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Ezrzh82spapOSyNvVFvXnqUW9eNnXRqwQSHmvkQc6kI">
> 
> lit. means: `“you share it, I kill you :)”` in french

😭😭😭😭 UGHHH WHYYYYY
WHY GOD WHYYY

**Comment by Heather95** [comment]
Hey @Anna-Hardy I think the tool you want already exists!
Try this: https://github.com/marcussacana/MwareStuff/releases
Here are the instructions: https://github.com/marcussacana/MwareStuff/blob/master/How%20To%20Use.md
Good luck!

**Comment by Anna-Hardy** [comment]
> Hey [@Anna-Hardy](https://github.com/Anna-Hardy) I think the tool you want already exists! Try this: https://github.com/marcussacana/MwareStuff/releases Here are the instructions: https://github.com/marcussacana/MwareStuff/blob/master/How%20To%20Use.md Good luck!

That's exactly what I was looking for! Thank you SO SO MUCH!!! 💗💗💗
